packages/core/contracts/UnirepSocial.sol

```diff
@@ -371,6 +409,11 @@ contract UnirepSocial is zkSNARKHelper {
     ) external payable {
         (,,,,uint maxReputationBudget,,,uint attestingFee,,) = unirep.config();
         uint256 voteValue = upvoteValue + downvoteValue;
+        // check if proof is submitted before
+        bytes32 proofNullifier = keccak256(
```

Comment:
we don't need nullifiers for proofs that are used in the unirep contract. We can take the non-subsidy proof nullifiers out if we wanted, but i guess it doesn't hurt to leave them.

I think in a newer version we can remove proof nullifiers in the unirep contract (except for ust) and let attesters handle it.

Reply:
yeah we can also use a subsidy nullifier to prevent this
can be in another PR